hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/util/TokenRelListCalculator.java

```diff
@@ -261,60 +265,60 @@ private TokenRelPointerUpdateResult updatePointersSurroundingTargetTokenRel(
      * @return the new head token number for the account
      */
-    private long calculateHeadTokenAfterDeletions(
-            final long currentHeadTokenNum,
+    private TokenID calculateHeadTokenAfterDeletions(
+            final TokenID currentHeadTokenId,
             @NonNull final Account account,
-            @NonNull final Map<Long, TokenRelation> updatedTokenRels,
-            @NonNull final Map<Long, TokenRelation> tokenRelsToDeleteByTokenId) {
+            @NonNull final Map<TokenID, TokenRelation> updatedTokenRels,
+            @NonNull final Map<TokenID, TokenRelation> tokenRelsToDeleteByTokenId) {
         final var accountId = BaseCryptoHandler.asAccount(account.accountNumber());
 
-        // Calculate the new head token number by walking the linked token rels until we find a token rel that is not in
+        // Calculate the new head token id by walking the linked token rels until we find a token rel that is not in
         // the list of token rels to delete
-        var currentTokenNum = currentHeadTokenNum;
+        var currentTokenId = currentHeadTokenId;
         // We use a safety counter to prevent infinite loops in case of a bug
         var safetyCounter = 0;
         TokenRelation currentWalkedTokenRel;
         do {
-            currentWalkedTokenRel = updatedTokenRels.containsKey(currentTokenNum)
-                    ? updatedTokenRels.get(currentTokenNum)
-                    : tokenRelStore.get(
-                            accountId,
-                            TokenID.newBuilder().tokenNum(currentTokenNum).build());
+            currentWalkedTokenRel = updatedTokenRels.containsKey(currentTokenId)
+                    ? updatedTokenRels.get(currentTokenId)
+                    : tokenRelStore.get(accountId, currentTokenId);
             if (currentWalkedTokenRel != null) {
-                if (!tokenRelsToDeleteByTokenId.containsKey(currentWalkedTokenRel.tokenNumber())) {
+                if (!tokenRelsToDeleteByTokenId.containsKey(currentWalkedTokenRel.tokenId())) {
                     // we found the first existing token rel that is not in the list of token rels to delete
                     break;
                 } else {
                     // we found a non-null token rel, but it is in the list of token rels to delete; we therefore
                     // continue walking the linked token rels
-                    currentTokenNum = currentWalkedTokenRel.nextToken();
+                    currentTokenId = currentWalkedTokenRel.nextToken();
                 }
             } else {
                 // We reached the end of the linked token rel pointers chain; there is no token rel that will qualify as
                 // the new head token number. We therefore set the new head token number to -1 and exit the do-while
                 // loop (since `currentWalkedTokenRel` is null)
-                currentTokenNum = -1;
+                currentTokenId = TokenID.newBuilder().tokenNum(-1).build();
```

Comment:
This will be null as well
```suggestion
                currentTokenId = null;
```

Reply:
same as above